src/structures/idt.rs

```diff
@@ -611,14 +614,27 @@ impl<F> Entry<F> {
     #[inline]
     fn set_handler_addr(&mut self, addr: u64) -> &mut EntryOptions {
         use crate::instructions::segmentation;
+        // SAFETY: CS segment is loaded directly from the register, so it's valid.
+        unsafe { self.set_handler_addr_unchecked(addr, segmentation::cs()) }
+    }
 
+    #[inline]
+    const unsafe fn set_handler_addr_unchecked(
```

Comment:
I think we should use a different name for this function to indicate that it takes a second argument. Maybe something like  `set_handler_addr_with_segment`?